client/src/components/SearchBar/LocationSearch/LocationSearch.js

```diff
@@ -0,0 +1,95 @@
+import React from 'react';
+import './LocationSearch.scss';
+import PlacesAutocomplete, {
+    geocodeByAddress,
+    getLatLng
+} from 'react-places-autocomplete';
+/* global google */
+
+/**
+ * Used to search the Monument coordinates field by location
+ */
+export default class LocationSearch extends React.Component {
+
+    constructor(props) {
+        super(props);
+        this.state = {
+            searchQuery: props.value,
+            sessionToken: ''
+        };
+    }
+
+    handleChange(newSearchQuery) {
+        const { sessionToken } = this.state;
+        if (!sessionToken) {
+            this.setState({sessionToken: new google.maps.places.AutocompleteSessionToken()})
+        }
+
+        // Google Maps API session tokens expire after 3 minutes
+        // If a request is made with an expired session token, Google charges the per-character rate
+        // Manually expire our local session token after 3 minutes so we don't get charged the higher price
+        window.setTimeout(() => {
+            this.setState({sessionToken: null});
```

Comment:
you might want to do something like this:

```
const sessionToken = this.state.sessionToken || new google.maps.places.AutocompleteSessionToken();
if (this.state.sessionToken !== sessionToken) {
    this.setState({sessionToken});
}

// Google Maps API session tokens expire after 3 minutes
// If a request is made with an expired session token, Google charges the per-character rate
// Manually expire our local session token after 3 minutes so we don't get charged the higher price
window.setTimeout(() => {
    if (sessionToken === this.state.sessionToken) {
        this.setState({sessionToken: null});
    }
}, 180000);
```

The reason being that this ensures that the sessionToken is only invalidated if no new token has been created in the mean time

Reply:
Done.